### DISCUSSION TITLE
Are x-data properties always reactive?

### DISCUSSION THREAD

**Discussion by docdunning** [alpinejs/alpine#4029] [1. Help]
I've encountered one or two situations where I'd like to define non-reactive properties, but as I understand it, anything I put in the x-data state is reactive by definition.  In Vue, by comparison, I have to use ref() to make an object reactive.

In the following (somewhat contrived) example, there are two div's with x-bind to set their style.  On the second div, I don't want the style to change, even if the bound property changes.  To achieve this, I've copied the value of the property to a regular JS variable outside the x-data scope.  By the way, I'm happy to see that the x-bind works with a variable outside x-data ... :-)

This works, and of course the copied value could be inside x-data, but then it would be reactive, and I want to avoid that if possible.  It makes the purpose more clear.

I guess I need something like x-bind:style.once, which would use the property to set the attribute but then drop the reactivity, or maybe to be able to define an x-data property that is not reactive.

I know this is a bit of an edge case, but is there some way to do this, that I've missed?  Maybe the approach I've taken in this example is the best way to do it ... ?

```
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Document</title>
  <script defer src="https://cdn.jsdelivr.net/npm/alpinejs@3.13.5/dist/cdn.min.js"></script>
  <script>
    let globalColor = ''
  </script>
</head>
<body>
<div x-data="{
  myColor: 'red',
  init() {globalColor = this.myColor}
}">

<div :style="{ color: myColor }">
  This should be <span x-text="myColor"></span>
</div>

<div :style="{ color: globalColor }">
  This should be <span x-text="globalColor"></span> and not change
</div>

<button @click="myColor = 'green'">Change color</button>

</div>

</body>
</html>
```

**Comment by ekwoka** [comment]
You could just make a separate value on the object itself that you don't update.

**Reply by docdunning** [reply]
Yes indeed, although I was looking for something that wasn't reactive at all.

But hey, this really is an ideal, and I'm OK with copying the reactive property off to somewhere else.

**Comment by ekwoka** [comment]
Could also just use x-init

`x-init="$el.style.color = color"`

Since x-init only runs once.
